### DISCUSSION TITLE
Enabling proper video player support in editor play-mode

### DISCUSSION THREAD

**Discussion by techanon** [vrchat-community/ClientSim#72] [Ideas]
I put together a few tooling scripts to enable AVPro in editor for play-mode.  
https://gitlab.com/techanon/videoplayershim
I was thinking about adding it as a reference in the videos documentation page https://docs.vrchat.com/docs/video-players but I thought maybe it would better serve existing in the clientsim repository so it's much more accessible to creators.  
I can put together a PR for this but I wanted some preliminary thoughts on it before I do.

Specifically it enables the following:
- When an AVPro Trial package is imported into the project, it will enable scripts on playmode enter that generates valid AVPro components in scene.
- Adds explicit support for YTDL resolver usage (via the provided VRChat ytdlp copy) for both AVPro and UnityVideo players.
  - This logic is mostly the same as what exists in some community video players, but modified slightly to include AVPro integration.

One of the caveats is that because AVPro does not exist within the Packages context as a UPM, so the scripts that depend on types from AVPro itself fail to compile. The tooling must exist within the Assets folder.

Is there a known way to automatically add assets into the Assets folder from a VPM/UPM repo?

**Comment by float3** [comment]
This would be nice to have!
